riscv-asm.md

```diff
@@ -26,19 +26,39 @@ and vector registers (V extension).
 
 ## General registers
 
-The RV32I base integer ISA includes 32 registers, named `x0` to `x31`. The
-program counter `PC` is separate from these registers, in contrast to other
-processors such as the ARM-32. The first register, `x0`, has a special function:
-Reading it always returns 0 and writes to it are ignored. As we will see later,
-this allows various tricks and simplifications.
+The **instruction set architecture** (ISA) of RISC-V contains 32 registers.
+There are two ways of viewing them: A hardware-centered approach and the
+programmer's view. 
 
-In practice, the programmer doesn't use this notation for the registers. Though
-`x1` to `x31` are all equally general-use registers as far as the processor is
-concerned, by convention certain registers are used for special tasks. In
-assembler, they are given standardized names as part of the RISC-V **application
-binary interface** (ABI). This is what you will usually see in code listings. If
-you really want to see the numeric register names, the `-M` argument to objdump
-will provide them.
+In the first case, the registers are named `x0` to `x31`. All are completely
+equal as general-use registers, except for `x0`, which has a special function:
+Reading it always returns 0 and writes to it are ignored. This allows various
+tricks and simplifications. The program counter `pc` is separate from these
+registers, in contrast to other processors such as the ARM-32. 
```

Comment:
I realize you didn't introduce the bit about ARM-32, but since you're digging around here, I'd suggest deleting the dependent clause.  We generally shouldn't make reference to other ISAs except when necessary, and it's not necessary here, since ARM is the outlier in this respect, not RISC-V.

Reply:
Sure!